DebugInfomationFormat.md

```diff
@@ -0,0 +1,220 @@
+#Debug Symbol Integration
+
+This is a design proposal for providing debug information within WebAssembly
+that would allow source-level debugging of code compiled into WebAssembly.
+
+## Goals and Principles
+
+The immediate goal is to implement enough debugging functionality in WebAssembly
+MVP to let users perform at least rudimentary debugging on their source.
+Another goal is not to bake in any limitations that would prevent reasonable
+future evolution of the standard.
+
+Specifically, we are currently not focused on identifying a language-independent
+subset of the debug format -- this proposal bundles language-dependent and
+language-independent parts together.
+
+META: We need an evolution strategy that allows new front-end/debugger pairs to
+use the format in the future to transfer information currently unanticipated.
+Examples: a) dynamic scoping in Lisp; b) full DWARF 4 equivalence.
+
+## Debug Info Goes into Extra Sections
+
+To allow easy stripping of debug info, all of it will go into a separate
+section.  In the binary format, this will be an unknown section inserted into
+the wasm file.  Because this new info will make the current
+[name section](BinaryEncoding.md#name-section) redundant, we propose deprecating
+the name section.
+
+META: How do we distinguish the debug-containing unknown section from any other
+unknown section?
+
+In the text format, this will be an extra child of the **module** element named
+**debug**.
+
+META: If there's a better name than **debug**, we can certainly use it.  Also,
+currently undecided about allowing multiple **debug** children, like **func**.
```

Comment:
This comment just says that I'm undecided whether we should allow multiple __debug__ children under __module__.  I'm leaning towards allowing multiple **debug**s as a syntactic convenience; their union will constitute the module's debug info.

As for the larger question of flexibility, I'd like this design to cover all debugging use cases supported in the MVP.  Beyond MVP, I envision extending the format a backward-compatible ways for each new use case.  But I don't pretend to know which use cases (and in which order) that will be -- LISP and DWARF are just some examples that I threw in to illustrate the possible variety of what could be coming one day.